perun/logic/commands.py

```diff
@@ -166,6 +168,42 @@ def init_perun_at(
     perun_log.minor_status(msg, status=f"{perun_log.path_style(perun_path)}")
 
 
+def try_init():
+    """Checks if the current instance is Perun and prompts user if he wishes to init
+
+    Currently, we assume, that if there is no perun and no git, we initialize the perun with empty git
+    and potentially with empty commit
+    """
+    try:
+        pcs.get_path()
+    except NotPerunRepositoryException:
+        # First we ask if we wish to init Perun
+        if common_kit.perun_confirm(
+            f"  - You are not in Perun instance. "
+            f"Do you wish to init empty {perun_log.highlight('Perun')}"
+            f" with {perun_log.highlight('git')}?"
+        ):
+            contains_git = GitRepository.contains_git_repo(os.getcwd())
+            init(os.getcwd(), vcs_type="git")
+
+            # Second we create empty commit
+            if not contains_git:
+                perun_log.minor_status(
+                    "Creating empty commit",
+                    status=perun_log.cmd_style('git commit --allow-empty -m "root"'),
+                )
+                try:
+                    external_commands.run_safely_external_command(
+                        'git commit --allow-empty -m "root"'
+                    )
+                    perun_log.minor_success("Creating empty commit")
+                except subprocess.CalledProcessError:
+                    perun_log.minor_fail("Creating empty commit")
```

Comment:
I know you always wanted to isolate all subprocess-related stuff to just one module. Should we perhaps import the currently used subprocess exceptions/errors in such module (I am not sure which module it is right now) and then internally import it from there?